klippy/extras/sht4x.py

```diff
@@ -0,0 +1,158 @@
+# SHT4x i2c temperature sensors
+#
+# Copyright (C) 2025 Milzo <overchecking@gmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+import logging
+from . import bus
+
+# SHT4x I2C Commands
+DEFAULT_ADDR = 0x44
+RESET = 0x94
+
+# Measurement command (high precision)
+MEASUREMENT_CMD = 0xFD
+
+# Measurement timing constant
+MEASUREMENT_TIME = 0.009  # 9ms
+RESET_TIME = 0.001  # 1ms
+
+class SHT4X:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.name = config.get_name().split()[-1]
+        self.reactor = self.printer.get_reactor()
+
+        # I2C setup
+        addr = config.getint('i2c_address', DEFAULT_ADDR)
+        self.i2c = bus.MCU_I2C_from_config(config, default_addr=addr)
+        self.mcu = self.i2c.get_mcu()
+
+        # Sensor state
+        self.temp = 0
+        self.humidity = 0
+        self._error_count = 0
+        self.sensor_ready = False
+
+        # Register with Klipper
+        self.printer.add_object("sht4x " + self.name, self)
+        self.printer.register_event_handler(
+            "klippy:connect", self.handle_connect)
+
+    def handle_connect(self):
+        """Initialize sensor"""
+        try:
```

Comment:
We probably want to start to fail if the sensor can't be initialized.
I can't imagine a situation where I enable the machine and want the sensor to be silently broken.
You are free to print to the logs, but you still need to raise error to the high level code.

```
self._error = self.i2c.get_mcu().error
```